### PR TITLE
You can now unbuckle others from the toolbox training machine with altclick.

### LINKED ISSUE #54796: You cannot unbuckle someone else's corpse or character from a Personnel Training Machine
## Round ID: [149659](https://scrubby.melonmesa.com/round/149659)

Is not round exclusive
## Reproduction:
This concerns /obj/structure/training_machine
If a player standing on one of these buckles themselves to it they are the only person who can unbuckle themselves. Another player can only remove someone from a training machine by destroying the training machine. This is an issue in particular with corpses since there is no way, short of destroying the object, to remove them from it. Clicking or alt clicking on the training machine or player does not cause them to un-buckle, nor does attaching a shooting target.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Title. (should have used the no whitespace parameter when making the commit smh)

## Why It's Good For The Game
This will close #54796.

## Changelog
:cl:
fix: You can now unbuckle others from the toolbox training machine with altclick.
/:cl:

### CODE REVIEW

**Comment by cacogen** [comment]
Yeah I had no idea you could buckle people to these

**Comment by brotherbeyondd** [comment]
you can buckle a nuggetted corpse and watch it run around

File: code/game/objects/structures/training_machine.dm [hunk 1]
```diff
@@ -174,6 +174,11 @@
 
 /obj/structure/training_machine/AltClick(mob/user)
 	. = ..()
+	if(!user.canUseTopic(src, BE_CLOSE, NO_DEXTERITY, NO_TK, FLOOR_OKAY))
+		return
```
Comment (LemonInTheDark) [inline code/game/objects/structures/training_machine.dm:178]: Should this apply to all altclicks?

Comment (Ghommie) [reply code/game/objects/structures/training_machine.dm:178]: This is needed to stop cyborgs from incapacitated people etc. from (remotely) unbuckling people when they shouldn't.

Comment (LemonInTheDark) [reply code/game/objects/structures/training_machine.dm:178]: I'm guessing this fixes some finagling around removing the item in the machines hand too?

Comment (Ghommie) [reply code/game/objects/structures/training_machine.dm:178]: Yea.

**Pull request merged**